.github/workflows/ci.yaml

```diff
@@ -7,27 +7,85 @@ on:
     branches:
       - main
 
+permissions:
+  checks: write
+  pull-requests: write
+  
 jobs:
   build:
     runs-on: ubuntu-22.04
-    container: fedora:38
     strategy:
       fail-fast: false
       matrix:
-        buildtype:
-          - debug
-          - release
+        buildtype: [ 'debug', 'release' ]
+        image: [ 'fedora:38', '' ]
```

Comment:
Yes, it's for Ubuntu.

`image` is used to specify the container image. For Ubuntu, we don't need a container, we can run the CI directly on the Ubuntu VM. That's why we have `''` for the image.

Reply:
This definitely needs a clarifying comment then

Reply:
We should use a container for Ubuntu too for consistency purpose, so all the build would be isolated in a container. If my understanding is correct, right now we build for Ubuntu directly on the host, then start a Fedora container to build on Fedora. Also, for now the check summary on the PR is empty when running on Ubuntu. 

<img width="458" alt="Screenshot 2023-09-05 at 11 33 29" src="https://github.com/facebook/bpfilter/assets/10826441/a13169c0-90fa-4a7b-908f-6cfdcd3a3f7a">

Reply:
I didn't use a container for Ubuntu because I thought that we could not run the integrated tests inside them.

Reply:
Integration tests having been run inside a container, so it might require more development to make them work, but it should be possible. Eventually, we'll be able to run integration tests in a container.